tools/test-backend

```diff
@@ -132,6 +143,23 @@ if __name__ == "__main__":
     else:
         suites = args
 
+    # While running --re-run, we access the file containing the
+    # details of the tests which failed in the last run of test-backend
+    # which is stored at zulip/var/last_test_failure.json and run the failed tests
+    # one by one until the errors are rectified
+    if options.re_run:
+        try:
+            with open(failed_test_path, 'r') as outfile:
+                failed_tests = ujson.load(outfile)
+                for test in failed_tests['name']:
+                    cmd = ['tools/test-backend', test]
+                    print('Running', " ".join(cmd))
+                    subprocess.check_call(cmd)
+        except (EnvironmentError, subprocess.CalledProcessError) as exception:
+            print('Got %s error while running the command, \
+                please rectify and try again.' % (exception))
+        sys.exit(0)
```

Comment:
This implementation is super inefficient (calling `test-backend` via subprocess for each test); instead, we should just treat it as though the arguments passed to `test-backend` was the list of tests in the file (I think it might work to just reset `args` a bit earlier in this file) and then just continue as normal.